Honor copy=true in directory mode for local file copies

ebudan opened the pull request:
This looks like an oversight: when getting local files, directory copies always result in a symlink. This may not be desirable in all use cases.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/go-getter?pullRequest=327)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/go-getter?pullRequest=327)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/go-getter?pullRequest=327).</sub>

sylviamoss commented on go.mod:
```diff
@@ -1,4 +1,4 @@
-module github.com/hashicorp/go-getter
+module github.com/ebudan/go-getter
```
```suggestion
module github.com/hashicorp/go-getter
```

ebudan replied:
See comments below

sylviamoss commented on get_file_unix.go:
```diff
@@ -41,11 +41,15 @@ func (g *FileGetter) Get(dst string, u *url.URL) error {
 
 	// Create all the parent directories
-	if err := os.MkdirAll(filepath.Dir(dst), g.client.mode(0755)); err != nil {
+	if err := os.MkdirAll(dst, g.client.mode(0755)); err != nil {
```
Why removing the `filepath.Dir`?

ebudan replied:
See comments below

sylviamoss reviewed (commented):
This is looking good 👍🏼 
I miss a test though. A test similar to:  https://github.com/hashicorp/go-getter/blob/main/get_file_test.go#L129

ebudan commented:
Added test as requested, and reverted my dir creation change - the reason for that was obscure and not useful here. Module name was changed as I had to use my fork - reverted, will need an additional fork unless this request works for you.

Two realizations during this commit: this leaves the Windows version uncovered (no opportunity here), and there is no support in the CLI tool for hard copy. What would be preferrable for the latter: a command line option (that only applies to copy... ouch), or a url parameter (go-getter file:///tmp/foo?copy=true /tmp/dst ...ouch)?

Smithx10 commented:
Merge this?
